cuqi/experimental/mcmc/_cwmh.py

```diff
@@ -0,0 +1,207 @@
+        :class:`ProposalBasedSamplerNew`.
+
+    Example
+    -------
+    .. code-block:: python
+        import numpy as np
+        import cuqi
+        # Parameters
+        dim = 5 # Dimension of distribution
+        mu = np.arange(dim) # Mean of Gaussian
+        std = 1 # standard deviation of Gaussian
+
+        # Logpdf function
+        logpdf_func = lambda x: -1/(std**2)*np.sum((x-mu)**2)
+
+        # Define distribution from logpdf as UserDefinedDistribution (sample
+        # and gradients also supported as inputs to UserDefinedDistribution)
+        target = cuqi.distribution.UserDefinedDistribution(
+            dim=dim, logpdf_func=logpdf_func)
+
+        # Set up sampler
+        sampler = cuqi.experimental.mcmc.CWMHNew(target, scale=1)
+
+        # Sample
+        samples = sampler.sample(2000).get_samples()
+
+    """
+    def __init__(self, target: cuqi.density.Density, proposal=None, scale=1,
+                 initial_point=None, **kwargs):
+        super().__init__(target, proposal=proposal, scale=scale,
+                         initial_point=initial_point, **kwargs)
+        # Set scale
+        if isinstance(scale, Number):
+            self.scale = np.ones(self.dim)*scale
+            self._scale_temp = np.ones(self.dim)*scale
+        elif isinstance(scale, np.ndarray):
+            self.scale = scale
+            self._scale_temp = scale.copy()
+
+        # set initial acceptance rate
+        self._acc = [np.ones((self.dim))]
+
+    def validate_target(self):
+        pass # All targets are valid
+
+    @ProposalBasedSamplerNew.proposal.setter 
+    def proposal(self, value):
+        fail_msg = "Proposal should be either None, "+\
+            f"{cuqi.distribution.Distribution.__class__.__name__} "+\
+            "conditioned only on 'location' and 'scale', lambda function, "+\
+            f"or {cuqi.distribution.Normal.__class__.__name__} conditioned "+\
+            "only on 'mean' and 'std'"
+
+        if value is None:
+            self._proposal = cuqi.distribution.Normal(
+                mean=lambda location: location,
+                std=lambda scale: scale,
+                geometry=self.dim,
+            )
+
+        elif isinstance(value, cuqi.distribution.Distribution) and sorted(
+            value.get_conditioning_variables()
+        ) == ["location", "scale"]:
+            self._proposal = value
+
+        elif isinstance(value, cuqi.distribution.Normal) and sorted(
+            value.get_conditioning_variables()
+        ) == ["mean", "std"]:
+            self._proposal = value(
+                mean=lambda location: location, std=lambda scale: scale
+            )
+
+        elif not isinstance(value, cuqi.distribution.Distribution) and callable(
+            value):
+            self._proposal = value
+
+        else:
+            raise ValueError(fail_msg)
+
+    def step(self):
+        """ Perform one step of the sampler by transitioning the current point
```

Comment:
I believe we can push this generic docstring to the base class. Other classes will inherit the docstring automatically.

Reply:
yes, that is great. I removed the doc string in step, tune, set_state and get_state.